### ISSUE TITLE
external request lane change not working properly

### ISSUE THREAD

**Issue by Joonki0111** [autowarefoundation/autoware.universe#8513]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

If moving to side lanes by [behavior_path_external_request_lane_change_module](https://github.com/autowarefoundation/autoware.universe/tree/release/v1.0/planning/behavior_path_external_request_lane_change_module), the module doesn't work as expected.

1. Path planned with both left and right path candidate as expected
![1](https://github.com/user-attachments/assets/91877b23-bbf5-4717-a5eb-3fd727d00727)

2. Ego triggered to change lane 
![2](https://github.com/user-attachments/assets/579f1609-3e1f-41f0-a8be-893f28f43760)

3. In middle of changing lane 
![3](https://github.com/user-attachments/assets/cdb859d2-fbb0-4ed1-9a2b-d7d6ed52145d)

4. Successfully changed lane with no available path candidate 
![4](https://github.com/user-attachments/assets/b044595b-92e2-4e5b-9e27-fd2253b37596)

    But in debug info, external_request_lane_change_left module can be found as candidate module
![5](https://github.com/user-attachments/assets/3067de95-92d3-427d-b336-79ba640c0f48)

At first I thought external request lane change in a row was impossible but found out that one side to another was possible.
In other words, lane change direction can't be shifted. 

### Expected behavior

change lane freely with [behavior_path_external_request_lane_change_module](https://github.com/autowarefoundation/autoware.universe/tree/release/v1.0/planning/behavior_path_external_request_lane_change_module)

### Actual behavior

Can't change direction of lane change

### Steps to reproduce

planning simulator with any map

### Versions

Ubuntu22.04 Humble 
Autoware release/v1.0

**Comment by zulfaqar-azmi-t4** [comment]
@Joonki0111 

Thank you so much for the report.
Is the goal in the middle lane?

The unlike normal LC, external request LC typically searches for non-preferred lane to use as target. If the goal is in the middle lane, it will become preferred lane, hence external request LC will not to be able to find its  target lane.

**Comment by Joonki0111** [comment]
@zulfaqar-azmi-t4 Thanks for the reply. I found the answer couple of hours ago which is same as your reply. I'm now currently working on modifying the /planning/mission_planning/route topic when external request LC is processed to change lane in any situation. Glad to hear it's not a bug.
